### PR TITLE
Fix #631: bun add throwing JSON lexer bug

### LINKED ISSUE #631: `bun add` throwing JSON Lexer bug
### Version

0.1.3

### Platform

Linux dev 5.15.52-1-lts #1 SMP Sat, 02 Jul 2022 20:04:03 +0000 x86_64 GNU/Linux

### What steps will reproduce the bug?

create any JS/TS project with the following jest config in the `package.json`
```json
{
  "...": "...",
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}
```

### How often does it reproduce? Is there a required condition?

it occurs anytime I run `bun add`

### What is the expected behavior?

Bun should just add a package like any other package managers, installing to the node_modules dir and also updating the package.json

### What do you see instead?

the following error is thrown:
```sh
bun add v0.1.3

error: Syntax Error!!

    "testRegex": ".*\.spec\.ts$",
                 ^
/...path.../Bun_NestJS/app/package.json:83:18 2534
SyntaxError parsing package.json in "/...path.../Bun_NestJS/app/"
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jwhear**
Fixes #631.  The string literal parsing code was not treating double backslash like the other escaped characters.

### CODE REVIEW

**Comment by jwhear** [comment]
Marking WIP as I have identified an issue where illegal JSON strings can cause a hang.

File: src/js_lexer.zig [hunk 1]
```diff
@@ -624,7 +624,7 @@ fn NewLexer_(
 
                         switch (lexer.code_point) {
                             // 0 cannot be in this list because it may be a legacy octal literal
-                            'v', 'f', 't', 'r', 'n', '`', '\'', '"', 0x2028, 0x2029 => {
```
Comment (Jarred-Sumner) [inline src/js_lexer.zig:627]: What if the string is `\\b`?  Or `\\u{2028}`? One where it does need to be escaped

Comment (jwhear) [reply src/js_lexer.zig:627]: I might be misunderstanding you, but `\\b` should be interpreted as "literal backlash, b".  This sequence is actually in the test case.  Similar logic for the Unicode sequence, it should be interpreted as a literal backslash followed by u, {, 2, etc.

**Comment by Jarred-Sumner** [comment]
Thank you. I understand now. It does change the transpiled output slightly.

This input:
```js
console.log('{"testRegex":"\\a\n\\b\\"}');
```

Bun v0.5.6:
```js
console.log(`{"testRegex":"\\a
\\b\\"}`);
```

Bun v0.5.7:
```js
console.log('{"testRegex":"\\a\n\\b\\"}');
```

Both evaluate to identical output for this scenario. 
```js
{"testRegex":"\a
\b\"}
```

**Pull request merged**